### ISSUE TITLE
About the results in the moleculenet paper

### ISSUE THREAD

**Issue by lihan97** [deepchem/deepchem#1711]
Hi,
I am very confused about that how the results in this paper were produced. For a random split dataset, is it just setting only one random seed or several (5 or 10) random seeds and then averaging?
Thank you!

**Comment by rbharath** [comment]
@miaecle Do you happen to remember how we handled the random seeds for the random split datasets?

**Comment by shenwanxiang** [comment]
Hi i think the author has fixed the random seed.

**Comment by lihan97** [comment]
> Hi i think the author has fixed the random seed.

Really? But it may be rigorous, the results with different random seeds could vary greatly

**Comment by lihan97** [comment]
I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
This file mentioned that:
Hyperparameters and all benchmark scripts for MoleculeNet are available at:
http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
I have two new question:
122,123,124 were used as the random seeds?
those split indices files in the documents are the benchmark splits?

**Comment by shenwanxiang** [comment]
> > Hi i think the author has fixed the random seed.
> 
> Really? But it may be not rigorous, the results with different random seeds could vary greatly

You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning

**Comment by shenwanxiang** [comment]
> I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> This file mentioned that:
> Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> I have two new question:
> 122,123,124 were used as the random seeds?
> those split indices files in the documents are the benchmark splits?

The MoleculeNet's dataset  

> > > Hi i think the author has fixed the random seed.
> > 
> > 
> > Really? But it may be not rigorous, the results with different random seeds could vary greatly
> 
> You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning

Many articles have been published, claiming that their own algorithms have gone a lot. In fact, if you look at their code, they use different random seeds to split their data sets. This is very unbelievable, almost falsified behavior.

**Comment by lihan97** [comment]
> > I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> > This file mentioned that:
> > Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> > http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> > Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> > I have two new question:
> > 122,123,124 were used as the random seeds?
> > those split indices files in the documents are the benchmark splits?
> 
> The MoleculeNet's dataset
> 
> > > > Hi i think the author has fixed the random seed.
> > > 
> > > 
> > > Really? But it may be not rigorous, the results with different random seeds could vary greatly
> > 
> > 
> > You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning
> 
> Many articles have been published, claiming that their own algorithms have gone a lot. In fact, if you look at their code, they use different random seeds to split their data sets. This is very unbelievable, almost falsified behavior.

Yeah, I found the same problem and I was really shocked at that! Actually, I'm doing some work on chemical modeling , so I'm trying to get those real benchmark dataset splits to make some comparisons with the results reported in MoleculeNet. I really hope the author can verify my answer or give me some suggestions.

**Comment by shenwanxiang** [comment]
> > > I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> > > This file mentioned that:
> > > Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> > > http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> > > Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> > > I have two new question:
> > > 122,123,124 were used as the random seeds?
> > > those split indices files in the documents are the benchmark splits?
> > 
> > 
> > The MoleculeNet's dataset
> > > > > Hi i think the author has fixed the random seed.
> > > > 
> > > > 
> > > > Really? But it may be not rigorous, the results with different random seeds could vary greatly
> > > 
> > > 
> > > You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning
> > 
> > 
> > Many articles have been published, claiming that their own algorithms have gone a lot. In fact, if you look at their code, they use different random seeds to split their data sets. This is very unbelievable, almost falsified behavior.
> 
> Yeah, I found the same problem and I was really shocked at that! Actually, I'm doing some work on chemical modeling , so I'm trying to get those real benchmark dataset splits to make some comparisons with the results reported in MoleculeNet. I really hope the author can verify my answer or give me some suggestions.

Hi, if you use the deepchem package, you can export the train, valid, and test dataset directly. 
The codes are in the paper of MoleculeNet, here is an example:

```
import deepchem as dc
tasks, datasets, transformers = dc.molnet.load_tox21(split='random', featurizer = 'Raw')
train, valid, test = datasets
train.x # smiles of training set
train.y #targets

```

**Comment by lihan97** [comment]
> > > > I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> > > > This file mentioned that:
> > > > Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> > > > http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> > > > Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> > > > I have two new question:
> > > > 122,123,124 were used as the random seeds?
> > > > those split indices files in the documents are the benchmark splits?
> > > 
> > > 
> > > The MoleculeNet's dataset
> > > > > > Hi i think the author has fixed the random seed.
> > > > > 
> > > > > 
> > > > > Really? But it may be not rigorous, the results with different random seeds could vary greatly
> > > > 
> > > > 
> > > > You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning
> > > 
> > > 
> > > Many articles have been published, claiming that their own algorithms have gone a lot. In fact, if you look at their code, they use different random seeds to split their data sets. This is very unbelievable, almost falsified behavior.
> > 
> > 
> > Yeah, I found the same problem and I was really shocked at that! Actually, I'm doing some work on chemical modeling , so I'm trying to get those real benchmark dataset splits to make some comparisons with the results reported in MoleculeNet. I really hope the author can verify my answer or give me some suggestions.
> 
> Hi, if you use the deepchem package, you can export the train, valid, and test dataset directly.
> The codes are in the paper of MoleculeNet, here is an example:
> 
> ```
> import deepchem as dc
> tasks, datasets, transformers = dc.molnet.load_tox21(split='random', featurizer = 'Raw')
> train, valid, test = datasets
> train.x #smiles list of training set
> train.y #targets
> ```
> 
> So I think maybe you don't need to know the random seeds in their paper. Just use this code for the generation of train, valid and test set.

Thanks for your suggestion. 
I have tried this method, but this function produces different splits each time. This function also randomly generate splits, instead of a fixed split.

**Comment by peastman** [comment]
If you first set the seed by calling `np.random.seed(seed)`, you should get deterministic results.  No idea what seed values were used in the paper though.

**Comment by shenwanxiang** [comment]
> > > > > I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> > > > > This file mentioned that:
> > > > > Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> > > > > http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> > > > > Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> > > > > I have two new question:
> > > > > 122,123,124 were used as the random seeds?
> > > > > those split indices files in the documents are the benchmark splits?
> > > > 
> > > > 
> > > > The MoleculeNet's dataset
> > > > > > > Hi i think the author has fixed the random seed.
> > > > > > 
> > > > > > 
> > > > > > Really? But it may be not rigorous, the results with different random seeds could vary greatly
> > > > > 
> > > > > 
> > > > > You‘re definitely right. So I think anyone who wants to make a comparison with DeepChem using their own algorithm, should use the same train, valid and test set, otherwise, there is no sense of meaning
> > > > 
> > > > 
> > > > Many articles have been published, claiming that their own algorithms have gone a lot. In fact, if you look at their code, they use different random seeds to split their data sets. This is very unbelievable, almost falsified behavior.
> > > 
> > > 
> > > Yeah, I found the same problem and I was really shocked at that! Actually, I'm doing some work on chemical modeling , so I'm trying to get those real benchmark dataset splits to make some comparisons with the results reported in MoleculeNet. I really hope the author can verify my answer or give me some suggestions.
> > 
> > 
> > Hi, if you use the deepchem package, you can export the train, valid, and test dataset directly.
> > The codes are in the paper of MoleculeNet, here is an example:
> > ```
> > import deepchem as dc
> > tasks, datasets, transformers = dc.molnet.load_tox21(split='random', featurizer = 'Raw')
> > train, valid, test = datasets
> > train.x #smiles list of training set
> > train.y #targets
> > ```
> > 
> > 
> > So I think maybe you don't need to know the random seeds in their paper. Just use this code for the generation of train, valid and test set.
> 
> Thanks for your suggestion.
> I have tried this method, but this function produces different splits each time. This function also randomly generate splits, instead of a fixed split.

Hi, I can get repeatable results using this code, without predefine the random seed in Numpy. I am using the version of deepchem 2.2.1.dev54: https://pypi.org/project/deepchem/

**Comment by shenwanxiang** [comment]
![image](https://user-images.githubusercontent.com/21102929/69624093-fa54b300-107e-11ea-8c5a-872a08961362.png)
![image](https://user-images.githubusercontent.com/21102929/69624123-0cceec80-107f-11ea-9f1d-9da497da5c79.png)

![image](https://user-images.githubusercontent.com/21102929/69624241-499ae380-107f-11ea-9f49-f7a50c03b017.png)

**Comment by shenwanxiang** [comment]
> I think I have got the answer. https://github.com/deepchem/deepchem/blob/master/examples/benchmark.py
> This file mentioned that:
> Hyperparameters and all benchmark scripts for MoleculeNet are available at:
> http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
> Then I found the splits in Hyperparameter_MoleculeNetv3.tar.gz.
> I have two new question:
> 122,123,124 were used as the random seeds?
> those split indices files in the documents are the benchmark splits?

I think these random seeds are just for running three times on GPUs. Although running on GPUs can not get repeatable results when you fix your random seed, but if fixed, the results are more similar

**Comment by shenwanxiang** [comment]
Repeat previous MolculeNet data split result

[00_prepare_dataset.pdf](https://github.com/deepchem/deepchem/files/4031480/00_prepare_dataset.pdf)
